E2E: fix issues with FSE: Smoke Test introduced with new Gutenberg version

worldomonation opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes #74009.
Related to https://github.com/Automattic/wp-calypso/pull/73979.

## Proposed Changes

This PR updates the steps involved in the `FSE: Smoke Test`.


## Testing Instructions

Ensure the following build configurations are passing:
  - [x] Calypso E2E (desktop)
  - [x] Calypso E2E (mobile)

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

worldomonation committed:
packages/calypso-e2e/src/lib/pages/full-site-editor-page.ts
- add new method `openEditor` to click on the preview pane on the right to load the editor

test/e2e/specs/fse/fse__temp-smoke-test.ts
- removed the skip step

worldomonation committed:
Add JSDoc.

worldomonation committed:
Remove commented out step

karenroldan reviewed (approved):
LGTM 🚀

worldomonation committed:
Extend timeout

jeyip commented:
Hmmm...it looks like the Calypso E2E (mobile) specs are still failing. Will take a close look after I finish reviewing a coworker's PR

worldomonation commented:
> Hmmm...it looks like the Calypso E2E (mobile) specs are still failing. Will take a close look after I finish reviewing a coworker's PR

Yes, it is failing because I slightly modified the test step to open the editor itself.

Unfortunately on mobile the page preview on right hand side is not present, so the steps I changed doesn't work.

jeyip commented:
> Unfortunately on mobile the page preview on right hand side is not present, so the steps I changed doesn't work.

Ah. Judging by the screenshots, this should be as easy as closing the navigation sidebar -- lemme see if I can dig up what I did a week ago

worldomonation committed:
packages/calypso-e2e/src/lib/pages/full-site-editor-page.ts
- add a new method to click on the buttons in the Site Editor Navigator screen

test/e2e/specs/fse/fse__temp-smoke-test.ts
- add new step to open the page template

worldomonation commented:
> > Unfortunately on mobile the page preview on right hand side is not present, so the steps I changed doesn't work.
> 
> ~Ah. Judging by the screenshots, this should be as easy as closing the navigation sidebar -- lemme see if I can dig up what I did a week ago~
> 
> ~Edit: https://github.com/Automattic/wp-calypso/pull/73465/files#diff-67c17b3bc0b1ef834201a5f1c92a3340eabbea6fd2d7b0dfaa63d759e439259aR33~
> 
> Edit 2: After looking at the context of this PR, I see what you're trying to accomplish. Feel free to ignore the advice above 😅 Will take a closer look soon

I think I have a working solution, pushing up.

jeyip commented:
> I think I have a working solution, pushing up.

Sounds good! It sounds like you're currently at a meetup, so if you'd like, I'm happy to shepherd the solution into production. Thanks again for responding to this so quickly

jeyip commented:
In a meeting. Shepherding this PR right after

jeyip reviewed (commented):
## Testing
### Requirements
- [x] test/e2e/specs/fse/fse__temp-smoke-test.ts passes Calypso E2E mobile specs
- [x] test/e2e/specs/fse/fse__temp-smoke-test.ts passes Calypso E2E desktop specs

### Browsers
- [x] Chrome

jeyip committed:
Remove commented out code

jeyip commented:
Waiting for CI builds to complete and then deploying the changes

worldomonation commented:
@jeyip Thanks for taking care of the deploy :)

jeyip commented:
Of course! Thanks for getting a fix out so quickly :D
